jobsdb/jobsdb_migrations.go

```diff
@@ -0,0 +1,228 @@
+package jobsdb
+
+import (
+	"database/sql"
+	"fmt"
+	"math"
+	"sync"
+
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+//MigrationState maintains the state required during the migration process
+type MigrationState struct {
+	sequenceProvider        SequenceProvider
+	dsForNewEvents          dataSetT
+	isDsForMigrationCreated bool
+	migrationDSCreationLock sync.RWMutex
+}
+
+/*
+SetupForImportAndAcceptNewEvents is used to initialize the HandleT structure.
+clearAll = True means it will remove all existing tables
+tablePrefix must be unique and is used to separate
+multiple users of JobsDB
+dsRetentionPeriod = A DS is not deleted if it has some activity
+in the retention time
+*/
+func (jd *HandleT) SetupForImportAndAcceptNewEvents(version int, isNew bool) {
+	jd.dsListLock.Lock()
+	defer jd.dsListLock.Unlock()
+	dsList := jd.getDSList(true)
+	newDSMin := int64(0)
+
+	if !isNew {
+		var minID, maxID sql.NullInt64
+		sqlStatement := fmt.Sprintf(`SELECT MIN(job_id), MAX(job_id) FROM %s`, dsList[len(dsList)-1].JobTable)
+		row := jd.dbHandle.QueryRow(sqlStatement)
+		err := row.Scan(&minID, &maxID)
+		jd.assertError(err)
+		if maxID.Valid {
+			newDSMin = int64(maxID.Int64)
+		} else {
+			panic("Unable to get max")
```

Comment:
Instead of using isNew, check dsList and if max/min is inValid, use that info to decide ds creation